syz-verifier/verifier.go

```diff
@@ -0,0 +1,264 @@
+	if !osutil.IsExist(runnerBin) {
+		log.Fatalf("bad syzkaller config: can't find %v", runnerBin)
+	}
+	execBin := cfg.ExecutorBin
+	if !osutil.IsExist(execBin) {
+		log.Fatalf("bad syzkaller config: can't find %v", runnerBin)
+	}
+
+	crashdir := filepath.Join(workdir, "crashes")
+	osutil.MkdirAll(crashdir)
+	for idx := range kis {
+		OS, Arch := target.OS, target.Arch
+		targetPath := OS + "-" + Arch + "-" + strconv.Itoa(idx)
+		osutil.MkdirAll(filepath.Join(workdir, targetPath))
+		osutil.MkdirAll(filepath.Join(crashdir, targetPath))
+	}
+
+	calls := make(map[*prog.Syscall]bool)
+	// TODO: move enabled system calls in the common config file
+	for _, id := range cfg.Syscalls {
+		calls[target.Syscalls[id]] = true
+	}
+
+	vrf := &Verifier{
+		crashdir:    crashdir,
+		kis:         kis,
+		target:      target,
+		choiceTable: target.BuildChoiceTable(nil, calls),
+		rnd:         rand.New(rand.NewSource(time.Now().UnixNano() + 1e12)),
+		runnerBin:   runnerBin,
+		executorBin: execBin,
+	}
+	vrf.prog = vrf.target.Generate(vrf.rnd, 2, vrf.choiceTable)
+
+	srv, err := startRPCServer(vrf)
+	if err != nil {
+		log.Fatalf("failed to initialise RPC server: %v", err)
+	}
+	vrf.srv = srv
+
+	for idx, ki := range kis {
+		go func(ki *KernelInfo, idx int) {
+			// TODO: support more VMs per VMPool
+			inst, err := ki.vmPool.Create(0)
+			if err != nil {
+				log.Fatalf("failed to create instance: %v", err)
+			}
+			defer inst.Close()
+
+			fwdAddr, err := inst.Forward(vrf.srv.port)
+			if err != nil {
+				log.Fatalf("failed to set up port forwarding: %v", err)
+			}
+
+			runnerBin, err := inst.Copy(vrf.runnerBin)
+			if err != nil {
+				log.Fatalf(" failed to copy runner binary: %v", err)
+			}
+			_, err = inst.Copy(vrf.executorBin)
+			if err != nil {
+				log.Fatalf("failed to copy executor binary: %v", err)
+			}
+			// TODO: Create RunnerCmd
+			cmd := fmt.Sprintf(".%s -name=%s -addr=%s -os=%s -arch=%s ", runnerBin, fmt.Sprintf("%d-%d", idx, 0), fwdAddr, vrf.target.OS, vrf.target.Arch)
+			outc, errc, err := inst.Run(ki.cfg.Timeouts.VMRunningTime, vrf.vmStop, cmd)
+			if err != nil {
+				log.Fatalf("failed to start runner: %v", err)
+			}
+
+			// TODO: use MonitorExecution in the future as that will create a proper
+			//  report, this should return just VM output for debugging purposes
+			for {
+				select {
+				case out, ok := <-outc:
+					if !ok {
+						outc = nil
+						continue
+					}
+					log.Printf("channel out: %s", out)
+				case err, ok := <-errc:
+					if !ok {
+						errc = nil
+						continue
+					}
+					log.Printf("error: %v", err)
+					// 	}
+					// }
+				}
+			}
+		}(ki, idx)
+	}
+
+	// hacky way to make sure we get the goroutine output
+	for {
+	}
+}
+
+func startRPCServer(vrf *Verifier) (*RPCServer, error) {
+	srv := &RPCServer{
+		vrf:     vrf,
+		runners: make(map[string]*Runner),
+	}
+	kis := vrf.kis
+	addr := kis[0].cfg.RPC
+	for idx := 1; idx < len(kis); idx++ {
+		if addr != kis[idx].cfg.RPC {
```

Comment:
I think it's better to combine with the loop in main that checks that other config values match.